docs/en/styling-and-theming/supplemental.md

```diff
@@ -0,0 +1,763 @@
+2.  The widget's parent class should be wrapped with `ThemedMixin`.
+3.  One or more of the widget's styling classes should be wrapped in a call to `this.theme()` when rendering the widget.
+
+By convention, there is a fourth requirement that is useful when developing widgets intended for distribution (this is a convention that widgets in Dojo's widget library follow):
+
+4.  The widget's root VDOM node - that is, the outer-most node rendered by the widget - should include a styling class named `root`. Doing so provides a predictable way to target the top-level node of a third-party themeable widget when overriding its styles in a custom theme.
+
+`ThemedMixin` and the `@theme()` decorator can be imported from the `@dojo/framework/widget-core/mixins/Themed` module. Using `ThemedMixin` on a widget provides access to the `this.theme()` function.
+
+### `ThemedMixin` `this.theme()` method
+
+```ts
+type SupportedClassName = string | null | undefined | boolean;
+```
+
+-   `this.theme(classes: SupportedClassName): SupportedClassName`
+-   `this.theme(classes: SupportedClassName[]): SupportedClassName[]`
+
+    -   Widgets can pass in one or multiple CSS class names and will receive back the corresponding amount of themed class names, suitable for passing on to the `classes` property when rendering the widget's WNodes or VNodes.
+
+        -   **Note 1:** Theme overrides are at the level of CSS classes only, not individual style properties within a class.
+
+        -   **Note 2:** If the currently active theme does **not** provide an override for a given styling class, the widget will fall back to using its default style properties for that class.
+
+        -   **Note 3:** If the currently active theme does provide an override for a given styling class, the widget will _only_ have the set of CSS properties specified in the theme applied to it. For example, if a widget's default styling class contains ten CSS properties but the current theme only specifies one, the widget will render with a single CSS property and lose the other nine that were not specified in the theme override.
+
+### `ThemedMixin` Widget Properties
+
+-   `theme` (optional)
+    -   If specified, [the provided theme](#writing-a-theme) will act as an override for any theme that the widget may use, and will take precedence over [the application's default theme](#making-themeable-applications) as well as [any other theme changes made in the application](#changing-the-currently-active-theme).
+-   `classes` (optional)
+    -   described in the [Passing extra classes to widgets](#passing-extra-classes-to-widgets) section.
+
+### Themeable Widget Example
+
+Given the following CSS module file for a themeable widget:
+
+> src/styles/MyThemeableWidget.m.css
+
+```css
+/* requirement 4, i.e. this widget is intended for wider distribution and therefore the main presentational styles for
```

Comment:
Manually wrapping this line at (or before?) "presentational" would reduce the likelihood of causing horizontal scrolling.